libs/wingc/src/lib.rs

```diff
@@ -78,11 +78,16 @@ pub unsafe extern "C" fn wingc_compile(ptr: u32, len: u32) {
 	if let Err(mut err) = results {
 		// Sort error messages by line number (ascending)
 		err.sort_by(|a, b| a.cmp(&b));
-		eprintln!(
-			"Compilation failed with {} errors\n{}",
+		let result = format!(
+			"Compilation failed with {} error(s)\n{}",
 			err.len(),
 			err.iter().map(|d| format!("{}", d)).collect::<Vec<_>>().join("\n")
 		);
+
+		let leaked = result.into_bytes().leak();
+		combine_ptr_and_length(leaked.as_ptr() as u32, leaked.len() as u32)
```

Comment:
This feels WASM only? Does this break native builds?

Reply:
This function is intentionally meant to only be used with WASM. There is an additional compile function that can be used by the native side with proper types.

Also with regards to breaking native builds, you may want to see [this PR](https://github.com/winglang/wing/pull/1453) if you haven't yet.